services/s3/src/main/java/software/amazon/awssdk/services/s3/internal/crossregion/utils/CrossRegionUtils.java

```diff
@@ -55,11 +62,17 @@ public static boolean isS3RedirectException(Throwable exception) {
     }
 
     private static boolean isRedirectError(S3Exception exceptionToBeChecked) {
-        int statusCode = exceptionToBeChecked.statusCode();
-        return statusCode == REDIRECT_STATUS_CODE || statusCode == TEMPORARY_REDIRECT_STATUS_CODE;
+        if (REDIRECT_STATUS_CODES.stream().anyMatch(status -> status.equals(exceptionToBeChecked.statusCode()))) {
+            return true;
+        }
+        if (getBucketRegionFromException(exceptionToBeChecked).isPresent()) {
+            return true;
+        }
```

Comment:
Which use case does this represent?

The malformed header is the next logical statement

Reply:
In this case, when the user directly calls "HeadBucket," we receive the "x-amz-bucket-region" in the error response. We shouldn't make another "HeadBucket" call. Instead, if we find "x-amz-bucket-region" in the S3Exception, we should use it to redirect the call to the region where the bucket actually exists.